src/Common/LRUFileCachePriority.h

```diff
@@ -0,0 +1,102 @@
+#pragma once
+
+#include <Common/IFileCachePriority.h>
+
+namespace DB
+{
+
+/// Based on the LRU algorithm implementation, the record with the lowest priority is stored at
+/// the head of the queue, and the record with the highest priority is stored at the tail.
+class LRUFileCachePriority : public IFileCachePriority
+{
+public:
+    using LRUQueue = std::list<FileCacheRecord>;
+    using LRUQueueIterator = typename LRUQueue::iterator;
+
+    class LRUFileCacheIterator : public IIterator
```

Comment:
it would probably look better (or at least more compact) if you make a forward declaration of LRUFileCacheIterator class here like
```
class LRUFileCachePriority
{
private: 
    class LRUFileCacheIterator; 
}

class LRUFileCachePriority::LRUFileCacheIterator  : public IIterator
{
}
```
as not small nested classes make it less readable